moodle_scanner/moodle_scanner.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python3
+import dataclasses
+import subprocess
+from typing import Any, Dict, List, Optional
+
+from artemis import load_risk_class
+from artemis.binds import Service, TaskStatus, TaskType
+from artemis.module_base import ArtemisBase
+from artemis.task_utils import get_target_url
+from karton.core import Task
+
+
+@dataclasses.dataclass
+class MoodleMessage:
+    category: str
+    problems: List[str]
+
+    @property
+    def message(self) -> str:
+        return f"{self.category}: {', '.join(self.problems)}"
+
+
+def process_moodle_json(result: Dict[str, Any]) -> List[MoodleMessage]:
+    messages: Dict[str, MoodleMessage] = {}
+
+    for key, value in result.items():
+        key_parts = key.replace("[", "").replace("]", "").split(". ")
+
+        if key in [
+            "[2. Moodle Security Checks]",
+            "[3. Deprecated Moodle Versions]",
+        ]:
+            continue
+
+        if len(key_parts) >= 2:
+            category = key_parts[1].capitalize()
+
+            if category.lower() != "info":
+                if isinstance(value, dict):
+                    for subkey, subvalue in value.items():
+                        if subvalue and subvalue not in [
+                            "Nothing to report, all seems OK!",
+                        ]:
+                            problem = f"{subkey} {subvalue}"
+                            if category not in messages:
+                                messages[category] = MoodleMessage(category=category, problems=[])
+                            messages[category].problems.append(problem)
+                elif isinstance(value, list):
+                    for item in value:
+                        if item and item not in [
+                            "Nothing to report, all seems OK!",
+                        ]:
+                            if category not in messages:
+                                messages[category] = MoodleMessage(category=category, problems=[])
+                            messages[category].problems.append(str(item))
+
+    return list(messages.values())
+
+
+@load_risk_class.load_risk_class(load_risk_class.LoadRiskClass.MEDIUM)
```

Comment:
how many requests does the scanner make? if e.g. a couple low-risk requests (homepage, readme etc.) we can treat it as low risk